shinysdr/plugins/wspr/demodulator.py

```diff
@@ -206,9 +218,11 @@ def fileClosed(self, filename):
 
         # wsprd expects its -f argument to be as if the recording was made with
         # a USB receiver
-        dial_freq = (self.__start_frequency - self.audio_frequency) / 1e6
+        dial_freq = (rf_frequency - self.audio_frequency) / 1e6
+        self.__decoder_active = defer.Deferred()
+        self.__decoder_active.addCallbacks(self.__decode_finished, self.__decode_finished)
```

Comment:
There's a `.addBoth` method that's shorthand for this.

But why use a deferred at all? Why not just pass a reference to the function you want called, or pass a reference to `self` and `WsprdProtocol` can call a method on that?